### PR TITLE
Interaction session for static reparenting

### PR DESCRIPTION
**Pull request by bkrmendy**
## Description 

This PR adds the necessary scaffolding for paste strategies. This PR itself doesn't introduce any new features, because
1. the paste strategies depend on https://github.com/concrete-utopia/utopia/pull/3778
2. the new interaction session can be tested the best when the actual paste strategies are in place.

The PR adds a placeholder strategy and a debug button to trigger it, and a way to set event listeners to commit the interaction session on `mousedown` and `keydown`.

### CODE REVIEW

File: editor/src/components/canvas/canvas-strategies/canvas-strategies.tsx [hunk 1]
```diff
@@ -152,6 +153,16 @@ const keyboardShortcutStrategies: MetaCanvasStrategy = (
   )
 }
 
+const staticReparentStrategies: MetaCanvasStrategy = (
+  canvasState: InteractionCanvasState,
+  interactionSession: InteractionSession | null,
+): Array<CanvasStrategy> => {
+  return mapDropNulls(
+    (factory) => factory(canvasState, interactionSession),
+    [testStaticReparentStrategy],
```
Comment (seanparsons) [inline editor/src/components/canvas/canvas-strategies/canvas-strategies.tsx:162]: Is it likely for there to be more values here? If not I'd just call `testStaticReparentStrategy` and default the result.

File: editor/src/components/canvas/controls/select-mode/select-mode-hooks.tsx [hunk 2]
```diff
@@ -954,6 +954,45 @@ export function useClearKeyboardInteraction(editorStoreRef: {
...
+      window.removeEventListener('mousedown', clearStaticReparentInteractionOnMouseDown)
+      if (
+        editorStoreRef.current.editor.canvas.interactionSession?.interactionData.type ===
+        'STATIC_REPARENT'
+      ) {
+        dispatch([CanvasActions.clearInteractionSession(true)], 'everyone')
+      }
+    }
+
+    window.addEventListener('mousedown', clearStaticReparentInteractionOnMouseDown, {
```
Comment (seanparsons) [inline editor/src/components/canvas/controls/select-mode/select-mode-hooks.tsx:972]: I think it's possible for these event listeners to leak and then potentially duplicate, we often use `useEffect` for these as we can return a cleanup function that calls `removeEventListener`.

**Review by seanparsons — CHANGES_REQUESTED** [review]
Other than the event listener handling this looks good!

File: editor/src/components/canvas/canvas-strategies/canvas-strategies.tsx (hunk 1 above)
Comment (bkrmendy) [reply editor/src/components/canvas/canvas-strategies/canvas-strategies.tsx:162]: I'll remove this in the next PR anyway so I don't think there's much use in polishing this

File: editor/src/components/canvas/controls/select-mode/select-mode-hooks.tsx (hunk 2 above)
Comment (bkrmendy) [reply editor/src/components/canvas/controls/select-mode/select-mode-hooks.tsx]: added a useEffect to clean the handlers up in https://github.com/concrete-utopia/utopia/pull/3786/commits/667daff4af8da84ad113f9f721d8a9035d31420e

**Review by enidemi — APPROVED** [review]

**Review by seanparsons — APPROVED** [review]
All of the thumbs up!

**Pull request merged**